### PR TITLE
Moving Proxy to Docker container and other bug fixes

### PR DESCRIPTION
**Pull request by mikeliddle**

### CODE REVIEW

File: CreateServer.ps1 [hunk 1]
```diff
@@ -439,6 +440,8 @@ Function New-TunnelEnvironment {
 
     New-Profiles
 
+    Remove-SSHRule
```
Comment (mikeliddle) [inline CreateServer.ps1]: Try...Finally

File: CreateServer.ps1 [hunk 2]
```diff
@@ -469,23 +472,26 @@ Function New-SprintSignoffEnvironment {
...
-    # Setup DNS
-    New-DnsServer
     # Setup WebServers
     Set-Endpoints
     Set-Content -Path "context.json" -Value (ConvertTo-Json $Context) -Force
     New-NginxSetup
 
     Update-RebootVM
 
+    Remove-SSHRule
```
Comment (mikeliddle) [inline CreateServer.ps1]: Try...Finally

File: scripts/TunnelHelpers.ps1 [hunk 3]
```diff
@@ -73,11 +73,13 @@ Function Set-Endpoints() {
         @{label= "Webapp"; url= "https://webapp.$($Context.TunnelFQDN)"},
         @{label= "Excluded"; url= "https://excluded.$($Context.TunnelFQDN)"},
         @{label= "WebappShort"; url= "https://webapp"},
+
```
Comment (mikeliddle) [inline scripts/TunnelHelpers.ps1]: Either add ExcludedShort or remove NL

File: scripts/TunnelHelpers.ps1 [hunk 4]
```diff
@@ -73,11 +73,13 @@ Function Set-Endpoints() {
...
         @{label= "Untrusted"; url= "https://untrusted.$($Context.TunnelFQDN)"},
         @{label= "IPAddressAPI"; url= "https://webapp.$($Context.TunnelFQDN)/api/IPAddress"},
         @{label= "ExcludedIPAPI"; url= "https://excluded.$($Context.TunnelFQDN)/api/IPAddress"},
         @{label= "Context"; url= "http://$($Context.VmName)-server.$($Context.Location).cloudapp.azure.com/context.json"},
```
Comment (mikeliddle) [inline scripts/TunnelHelpers.ps1]: Remove Context, rename Internal to just Context

File: scripts/createCerts.sh [hunk 5]
```diff
@@ -152,6 +152,8 @@ CreateUserCert() {
         -CAcreateserial -out certs/user.pem -extensions req_ext -extfile user.conf >> certs.log 2>&1
     openssl pkcs12 -export -out private/user.pfx -inkey private/user.key -in certs/user.pem -passout pass: >> certs.log 2>&1
```
Comment (mikeliddle) [inline scripts/createCerts.sh:153]: give a static passphrase

Comment (mikeliddle) [reply scripts/createCerts.sh:153]: maybe need to rework the whole call

Comment (mikeliddle) [reply scripts/createCerts.sh:153]: Addressing in future PR

**Review by toddbohm — APPROVED** [review]

**Pull request merged**